cryptography/src/bls12381/primitives/group.rs

```diff
@@ -319,6 +293,14 @@ impl Scalar {
     }
 
     /// Creates a new scalar from the provided index (a scalar offset by 1).
+    ///
+    /// This is important, because when doing polynomial secret sharing,
+    /// `f(0)` is the secret, and `f(x_0), f(x_1), ...` are the shares, for
+    /// particular points `x_i`.
+    ///
+    /// One simple method of choosing the points is this method, which
+    /// maps `0` to `1`, `1` to `2`, etc. ensuring that none of the points
+    /// collide with `0`, which would reveal the secret.
     pub fn from_index(i: u32) -> Self {
```

Comment:
I guess my only real feedback at this point is whether or not this should be part of `Mode` (more or less replacing our call to `from_index` to be `from_u64` with this 1 offset and comment.)

Reply:
I think removing this makes sense, yeah.